lib/azure/armrest/model/base_model.rb

```diff
@@ -62,7 +64,18 @@ def initialize(json)
           @json = json
         end
 
-        @ostruct = OpenStruct.new(hash)
+        ostruct = OpenStruct.new(hash)
+        keys = ostruct.to_h.keys
+
+        # If the method already exists then create an "_alias" method for it.
+        methods.each do |m|
+          if keys.include?(m)
+            ostruct.class.instance_eval{ alias_method "_#{m}", m }
+            instance_eval{ define_singleton_method("_#{m}"){ ostruct[m] } }
```

Comment:
Actually my above example has problems too. We should not traverse `obj.methods(false)`, instead we should traverse `obj.to_h.keys` similar to your code.

Your original fix does not handle the case that the key is in camel format such as `toS`.